### PR TITLE
add initial `embedded-hal` v1 implementations for pins, delays & PWM

### PR DESCRIPTION
**Pull request by rursprung**
this is a first step towards implementing `embedded-hal` v1 traits.

see the individual commit messages for further details.

this is as far as i'll take it in terms of implementation, as i only have an arduino uno lying around and currently don't have any peripherals to test with. however, this is a starting point for others to add the trait implementations which they'll need.

namely the SPI & I2C traits are not implemented. all other traits from `embedded-hal` should be implemented with this PR. note that some traits from v0 were moved to other crates (e.g. `embedded-io`) or were removed in v1.

see also #468

### CODE REVIEW

**Comment by Rahix** [comment]
Hi,

sorry, I was too fast about merging the dependabot PRs :(  Can you please rebase your patch series ontop of latest `main` and resolve the conflicts?

**Comment by rursprung** [comment]
> Hi,
> 
> sorry, I was too fast about merging the dependabot PRs :( Can you please rebase your patch series ontop of latest `main` and resolve the conflicts?

no worries. i've rebased & pushed it again.

File: avr-hal-generic/Cargo.toml [hunk 1]
```diff
@@ -11,9 +11,11 @@ ufmt = "0.2.0"
 paste = "1.0.0"
 avr-device = "0.5.3"
 embedded-storage = "0.2"
+embedded-hal = "1.0.0-rc.3"
 
-[dependencies.embedded-hal]
+[dependencies.embedded-hal-v0]
 version = "0.2.3"
+package = "embedded-hal"
```
Comment (Rahix) [inline avr-hal-generic/Cargo.toml:18]: Thanks for doing it this way, it makes the transition so much easier :)

**Review by Rahix — APPROVED** [review]
Looking very good indeed, thanks so much for picking up work on this.  And thanks for the clean commits, it is much appreciated :)

Let's merge this as is and continue iterating on the support for the new e-h in `main`.

**Pull request merged**